include/oboe/ResultWithValue.h

```diff
@@ -40,6 +40,21 @@ class ErrorOrValue {
         return mValue;
     }
 
+    /**
+     * Checks if the result was OK. Example usage:
+     *
+     * <code>
+     *     if (result.ok()) {
+     *         printf("Operation was successful");
+     *     } else {
+     *         printf("Operation was unsuccessful. Error: %s", convertToText(result.error()));
+     *     }
+     * </code>
+     *
+     * @return true if OK, false if error
+     */
+    bool ok() const { return mError == Result::OK; }
```

Comment:
Could you please consider making it possible to compare `ResultWithValue` against `Result` directly? This way, both `Result` and `ResultWithValue` return types can be handled in the same way in client code.

Currently you have code fragments like this:

    ResultWithValue setBufferSizeResult = mAudioStream->setBufferSizeInFrames(...)
    if (!setBufferSizeResult.ok()) { ... }
    result = mAudioStream->requestStart();
    if (result != Result::OK){ ... }

It would be much more consistent if it look like this instead:

    auto setBufferSizeResult = mAudioStream->setBufferSizeInFrames(...)
    if (setBufferSizeResult != Result::OK) { ... }
    auto result = mAudioStream->requestStart(); 
    if (result != Result::OK){ ... }

Reply:
Need comparison operator overload.

Reply:
As I've explained, it's sufficient to provide a conversion operator into the primitive `Result` type. The compiler will use it for coercing `ResultWithValue` into `Result` for performing the comparison. That's easier than providing overloads for all comparison operators.

Reply:
Great suggestion, I've implemented it and it feels much better in use 👍 

I've also removed the now redundant `ok()` method.